### PR TITLE
feat: Add dropdown menu for profiles

### PR DESCRIPTION
**Pull request by Jan200101**
add dropdown menu for profiles

### CODE REVIEW

**Comment by Jan200101** [comment]
@GeckoEidechse

**Comment by Jan200101** [comment]
Had to fix a small issue where the first commit did not have a missing import (how this evaded me is unknown, `tauri dev` didn't error on this)

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
In quick test, switching profile does not updated the detected version in PlayView.

File: src-tauri/src/northstar/profile.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    "runtime/",
+    "save_data/",
+    "Northstar.dll",
+    "enabledmods.json",
+    "placeholder.playerdata.pdata",
+    "LEGAL.txt",
+];
+
+#[tauri::command]
+pub fn fetch_profiles(game_install: GameInstall) -> Result<Vec<String>, String> {
```
Comment (GeckoEidechse) [inline src-tauri/src/northstar/profile.rs:31]: I'd add some [short doc](https://doc.rust-lang.org/reference/comments.html#doc-comments) comment to the function.

File: src-tauri/src/northstar/profile.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                profiles.push(profile_name);
+            }
+        }
+    }
+
+    Ok(profiles)
+}
+
+#[tauri::command]
+pub fn validate_profile(game_install: GameInstall, profile: String) -> bool {
```
Comment (GeckoEidechse) [inline src-tauri/src/northstar/profile.rs:60]: ditto

File: src-vue/src/views/SettingsView.vue [hunk 3]
```diff
@@ -172,6 +199,35 @@ export default defineComponent({
...
+            let state = store.state;
+
+            await invoke("validate_profile", { gameInstall: state.game_install, profile: value })
+                .then(async (message) => {
+                    if (!message)
+                    {
+                        // Profile is no longer valid, inform the user...
+                        showErrorNotification(
+                            i18n.global.tc('notification.profile.invalid.text'),
+                            i18n.global.tc('notification.profile.invalid.title')
```
Comment (GeckoEidechse) [inline src-vue/src/views/SettingsView.vue:214]: These two translations are missing from `en.json` btw ^^

**Comment by GeckoEidechse** [comment]
UI wise, there should maybe also be some explainer of what profiles are. Maybe a small `?` icon next to `Active Profile` that if hovered/clicked shows a tooltip or reveals small text that gives a short explainer.

Personally I'd still suggest to gate profile selecting behind DevView for now so that we can skip all these small UI quirks to that are kinda needed to make the feature "end-user compatible" and so that we can already test it with contributors.

Then we could even do a 3.0 release where we basically just move it from DevView to RepairView when deemed stable enough ^^

**Comment by Jan200101** [comment]
> In quick test, switching profile does not updated the detected version in PlayView.

I had only picked the first commit, as you requested.
The commits that fix that very cherry picked onto this branch.

**Comment by Jan200101** [comment]
Validated that the verison is now updated when changing the profile by switching to R2Ronin (which FlightCore cannot detect but is still a valid Profile)

**Comment by Jan200101** [comment]
> UI wise, there should maybe also be some explainer of what profiles are.

Future:tm:

> Personally I'd still suggest to gate profile selecting behind DevView

Done

**Review by GeckoEidechse — APPROVED** [review]
Confirmed working on Windows and Linux (skipped actually launching the game on Linux cause no working Titanfall2 install)

Requested code changes have been addressed and there isn't anything major I'd change with this PR rn. We could move the `SKIP_PATHS` and `MAY_CONTAIN` to `constant.rs` but I don't really see too much of a need to do so rn ^^

So good to merge from my end :D

**Comment by Jan200101** [comment]
based on #444

**Pull request merged**